guide/popular-topics/intents.md

```diff
@@ -0,0 +1,104 @@
+# Gateway Intents
+
+Gateway Intents were introduced to the library in v12 and allow you to pick-and-choose which events you wish your bot to receive and/or omit.
+Intents are groups of pre-defined events that the discord.js client will conditionally subscribe to. For example, omitting the `GUILD_PRESENCES` intent would prevent the discord.js client from receiving any member-related presence events. Before this feature, the discord.js client would receive all events, thus consuming memory for data you may not be using.
+
+:::warning
+Intents are optional on the v6 gateway but will become mandatory in a future gateway version.
+:::
+
+<branch version="11.x">
+
+Intents are not available in version 11, please update to version 12 of the library if you want to use gateway intents in your bot.
+
+</branch>
+
+<branch version="12.x">
+
+## Enabling Intents
+
+You can choose which intents you'd like to receive as client options when instantiating your bot client.
```

Comment:
Intents in discord.js are built on top of the BitField wrapper
At least a short introduction to computing custom intents should be included in this guide (does not need to be as in-depth as the permissions guide, of course) - the computing starting from <https://discord.js.org/#/docs/main/stable/class/Intents?scrollTo=s-ALL> preferably

Good intends to use for an example here are "DIRECT_MESSAGE_TYPING" and "GUILD_MESSAGE_TYPING" as a lot of bots don't have any typing functionality and this is a rather big impact we already recommended with disabling events for people wanting to cut memory usage down